src/controllers/breaches.js

```diff
@@ -1,13 +1,319 @@
 import { mainLayout } from '../views/layouts/main.js'
 import { breaches } from '../views/partials/breaches.js'
-
-function breachesPage (req, res) {
+import { setBreachResolution, setBreachesResolved, updateBreachStats, getUserEmails } from '../db/index.js'
+import { getBreachesForEmail, filterBreaches, getSha1 } from '../utils/index.js'
+async function breachesPage (req, res) {
+  // TODO: remove: to test out getBreaches call with JSON returns
+  const breachesData = await getAllEmailsAndBreaches(req.user, req.app.locals.breaches)
   const data = {
     locale: req.appLocale,
+    breachesData,
     partial: breaches
   }
 
   res.send(mainLayout(data))
 }
 
-export { breachesPage }
+/**
+ * Get breaches from the database and return a JSON object
+ * TODO: Takes in additional query parameters:
+ *
+ * status: enum (resolved, unresolved)
+ * email: string
+ * @param {object} req
+ * @param {object} res
+ */
+async function getBreaches (req, res) {
+  const allBreaches = req.app.locals.breaches
+  const sessionUser = req.user
+  const resp = await getAllEmailsAndBreaches(sessionUser, allBreaches)
+  return res.json(resp)
+}
+
+/**
+ * Modify breach resolution for a user
+ * @param {object} req containing {user, body: {affectedEmail, recencyIndex, resolutionsChecked, isUnresolve}}
+ *
+ * recencyIndex: corresponds to the relevant breach from HIBP
+ *
+ * resolutionsChecked: has the following structure [DataTypes]
+ *
+ * @param {object} res JSON object containing the updated breach resolution
+ */
+async function putBreachResolution (req, res) {
```

Comment:
@toufali Ignore this function, I will create another PR addressing the PUT call. I want to get the review started since it's already a pretty huge PR